astropy/coordinates/representation.py

```diff
@@ -43,15 +44,17 @@ def __init__(cls, name, bases, dct):
 
         REPRESENTATION_CLASSES[cls.get_name()] = cls
 
-p_opt = np.get_printoptions()
-def _fstyle(x):
-    fmt_str = "{0:." + str(p_opt['precision']) + "f}"
-    s = fmt_str.format(x)
-    s_trunc = s.rstrip("0")
-    if s_trunc.split(".")[1] == "":
-        return s_trunc + "0"
+
+def _fstyle(precision, x):
+    fmt_str = '{0:.{precision}f}'
+    s = fmt_str.format(x, precision=precision)
+    s_trunc = s.rstrip('0')
+    if s_trunc[-1] == '.':
+        # Ensure there is one trailing 0 after a bare decimal point
+        return s_trunc + '0'
     else:
-        return str(x)
+        return s_trunc
```

Comment:
The main bug was right here.  This function wasn't actually returning the truncated format string in many cases.